src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateEnumActionsUtils.java

```diff
@@ -0,0 +1,236 @@
+     * @param changeOrderEnumCallback the function that is called to apply the change in the order.
+     * @param <T>                     the enum type of the elements to change the order for.
+     * @return an optional update action if the the order of the enum values is not identical.
+     *         Otherwise, if the enum values order is identical, an empty optional is returned.
+     */
+    @Nonnull
+    public static <T extends WithKey> Optional<UpdateAction<ProductType>> buildChangeEnumValuesOrderUpdateAction(
+        @Nonnull final String attributeDefinitionName,
+        @Nonnull final List<T> oldEnumValues,
+        @Nonnull final List<T> newEnumValues,
+        @Nonnull final BiFunction<String, List<T>, UpdateAction<ProductType>> changeOrderEnumCallback) {
+
+        final List<String> newKeys = newEnumValues
+            .stream()
+            .map(WithKey::getKey)
+            .collect(Collectors.toList());
+
+        final List<String> existingKeys = oldEnumValues
+            .stream()
+            .map(WithKey::getKey)
+            .filter(newKeys::contains)
+            .collect(Collectors.toList());
+
+        final List<String> notExistingKeys = newKeys
+            .stream()
+            .filter(newKey -> !existingKeys.contains(newKey))
+            .collect(Collectors.toList());
+
+        final List<String> allKeys = Stream.concat(existingKeys.stream(), notExistingKeys.stream())
+                                           .collect(Collectors.toList());
+
+
+        return buildUpdateAction(
+            allKeys,
+            newKeys,
+            () -> changeOrderEnumCallback.apply(attributeDefinitionName, newEnumValues)
+        );
+    }
+
+    /**
+     * Checks if there are any new enum values which are not existing in the {@code oldEnumValues}.
+     * If there are, then "add" enum values update actions are built.
+     * Otherwise, if there are no new enum values, then an empty list is returned.
+     *
+     * @param attributeDefinitionName the attribute definition name whose enum values belong to.
+     * @param oldEnumValues           the list of olf enum values.
+     * @param newEnumValues           the list of new enum values.
+     * @param addEnumCallback         the function that is called in order to add the new enum instance
+     * @param <T>                     the enum type of the element to add.
+     * @return a list of enum values update actions if there are new enum value that should be added.
+     *         Otherwise, if the enum values are identical, an empty optional is returned.
+     */
+    @Nonnull
+    public static <T extends WithKey> List<UpdateAction<ProductType>> buildAddEnumValuesUpdateActions(
+        @Nonnull final String attributeDefinitionName,
+        @Nonnull final List<T> oldEnumValues,
+        @Nonnull final List<T> newEnumValues,
+        @Nonnull final BiFunction<String, T, UpdateAction<ProductType>> addEnumCallback) {
```

Comment:
It has to be the parent class because the action can be `AddEnumValue` or `AddLocalizedEnumValue` depending on the addEnumCallback will return one or the other. 

Another thing is that I can treat all update action as a more generic `UpdateAction<ProductType>` type, so then all update actions applied to the product type are the same and easier to manipulate in more abstract methods used at higher levels of the product type sync